homeassistant/components/shelly/cover.py

```diff
@@ -199,19 +202,42 @@ def is_opening(self) -> bool:
         """Return if the cover is opening."""
         return cast(bool, self.status["state"] == "opening")
 
+    def launch_update_task(self) -> None:
+        """Launch the update position task if needed."""
+        if not self._update_task:
+            self._update_task = (
+                self.coordinator.config_entry.async_create_background_task(
+                    self.hass, self.update_position(), "Shelly cover update"
+                )
+            )
+
+    async def update_position(self) -> None:
+        """Update the cover position every second."""
+        await self.coordinator.device.update_status()
+        self.async_write_ha_state()
+        await asyncio.sleep(RPC_COVER_UPDATE_TIME_SEC)
+        while self.is_closing or self.is_opening:
+            await self.coordinator.device.update_status()
+            self.async_write_ha_state()
+            await asyncio.sleep(RPC_COVER_UPDATE_TIME_SEC)
+        self._update_task = None
+
     async def async_close_cover(self, **kwargs: Any) -> None:
         """Close cover."""
         await self.call_rpc("Cover.Close", {"id": self._id})
+        self.launch_update_task()
```

Comment:
This approach will only work when the trigger is started from HA, but not from the device or anything external to Home Assistant.
You can override the update callback and so that you can act on state changes:
https://github.com/home-assistant/core/blob/473709172204f1fec2fe3287ddad3cda8d7fbc56/homeassistant/components/shelly/entity.py#L427
and start the task when the cover is closing or opening